File: solutions/2025/day01/part1.py
def solve(input: str) -> int:
    pos, zeros = 50, 0
    for line in input.strip().split('\n'):
        delta = int(line[1:]) * (-1 if line[0] == 'L' else 1)
        pos = (pos + delta) % 100
        zeros += pos == 0
    return zeros

File: solutions/2025/day01/part1.ts
export function solve(input: string): number | string {
  const lines = input.trim().split("\n");

  const rotations = lines.map((it) => toInt(it));

  let value = 50;
  let password = 0;

  for (const rotation of rotations) {
    value = (100 + (value + rotation)) % 100;
    if (value === 0) {
      password += 1;
    }
  }

  return password;
}

function toInt(line: string) {
  return parseInt(line.replace("L", "-").replace("R", "+"), 10);
}
